.github/actions/setup-environment/action.yml

```diff
@@ -0,0 +1,24 @@
+name: 'setup environment'
+description: 'setup go environment'
+
+inputs:
+  GO_VERSION:
+    description: 'Version of golang to setup'
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Check out the codebase.
+      uses: actions/checkout@v4
```

Comment:
Won't be the caller doing the checkout? Or can we remove the checkout from the caller?

Reply:
Good point. I'll remove from here, my understanding is the caller should have it.